File: Data Structures & Algorithms/validate-parentheses/submission-0.py
class Solution:
    def isValid(self, s: str) -> bool:
        paranthesis = list()
        match = dict()
        match["}"] = "{"
        match["]"] = "["
        match[")"] = "("
        for i in s:
            if i in ["{", "(", "["]:
                paranthesis.append(i)
            else:
                if ((len(paranthesis) ==0) or match[i] != paranthesis[-1]):
                    return False
                else: 
                    paranthesis.pop()
        if (len(paranthesis) == 0):
            return True
        else:
            return False

File: Data Structures & Algorithms/k-closest-points-to-origin/submission-0.cpp
class Solution {
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        vector<vector<int>> p;
        std::priority_queue<pair <double, vector<int> >, vector<pair<double, vector<int> >>, std::greater<pair<double, vector<int>>>> pq;
        for (auto point: points) {
            double distance = (point[0]*point[0]) + (point[1]*point[1]);
            pq.push({distance, point});
        }
        while(k--) {
            auto pnt = pq.top();
            p.push_back(pnt.second);
            pq.pop();
        }
        
        return p;
    }
};
